src/pages/task/TestsTableCore.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useEffect } from "react";
+import { ColumnProps, TableProps } from "antd/es/table";
+import { InfinityTable } from "antd-table-infinity";
+import { msToDuration } from "utils/string";
+import { Tag, Spin } from "antd";
+import { TESTS_QUERY } from "utils/gql/queries";
+import { useParams, useLocation, useHistory } from "react-router-dom";
+import { useQuery } from "@apollo/react-hooks";
+import { usePrevious } from "utils/hooks";
+import {
+  Categories,
+  RequiredQueryParams,
+  Sort,
+  ValidInitialQueryParams,
+  Limit
+} from "pages/task/types";
+import get from "lodash.get";
+import queryString from "query-string";
+
+interface TaskTests {
+  id: string;
+  status: string;
+  testFile: string;
+  duration: number;
+}
+
+interface UpdateQueryArg {
+  taskTests: [TaskTests];
+}
+
+const loadMoreContent = (): JSX.Element => (
+  <div
+    style={{
+      textAlign: "center",
+      paddingTop: 40,
+      paddingBottom: 40,
+      border: "1px solid #e8e8e8"
+    }}
+  >
+    <Spin tip="Loading..." />
+  </div>
+);
+const columns: Array<ColumnProps<TaskTests>> = [
+  {
+    title: "Name",
+    dataIndex: "testFile",
+    key: Categories.TestName,
+    sorter: true
+  },
+  {
+    title: "Status",
+    dataIndex: "status",
+    key: Categories.Status,
+    sorter: true,
+    render: (tag: string): JSX.Element => {
```

Comment:
i think the return type will be automatically inferred from the return value. you might not need to assert the return type in the function declaration.